custom/ilsgateway/comparison_reports.py

```diff
@@ -204,3 +206,46 @@ def rows(self):
                     reverse_lazy('product_availability_delete', kwargs={'domain': self.domain, 'pk': element.pk})
                 )
             ]
+
+
+class SupplyPointStatusReport(GenericTabularReport, DatespanMixin,
+                              CustomProjectReport, ProjectReportParametersMixin):
+    name = 'Supply Point Status Report'
+    slug = 'supply_point_status'
+    fields = [AsyncLocationFilter, DatespanFilter]
+    emailable = False
+    exportable = True
+
+    @classmethod
+    def show_in_navigation(cls, domain=None, project=None, user=None):
+        return False
+
+    @property
+    def location_id(self):
+        return self.request.GET.get('location_id', '')
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn('Location'),
+            DataTablesColumn('Type'),
+            DataTablesColumn('Value'),
+            DataTablesColumn('Date'),
+        )
+
+    @property
+    def rows(self):
+        locations = get_object_or_404(
+            SQLLocation, location_id=self.location_id
+        ).get_descendants(include_self=True)
+        data = SupplyPointStatus.objects.filter(
+            location_id__in=list(locations.values_list('location_id', flat=True)),
+            status_date__range=[self.datespan.startdate, self.datespan.enddate]
+        ).order_by('-status_date')
+        for element in data:
+            yield [
+                SQLLocation.objects.get(location_id=element.location_id).name,
```

Comment:
this would be good to call out to a `quickcache`d function that returns the name and keeps it in the cache for a while (maybe 30 minutes or something?) since they change very rarely.